skimage/measure/tests/test_regionprops.py

```diff
@@ -529,6 +529,31 @@ def test_column_dtypes_complete():
     assert set(COL_DTYPES.keys()).union(OBJECT_COLUMNS) == set(PROPS.values())
 
 
+def test_column_dtypes_correct():
+    region = regionprops(SAMPLE, intensity_image=INTENSITY_SAMPLE)[0]
+    for col in COL_DTYPES:
+        r = region[col]
+
+        if col in OBJECT_COLUMNS:
+            assert COL_DTYPES[col] == object
+            continue
+
+        if np.isscalar(r):
+            t = type(r)
+        else:
+            if np.isscalar(r[0]):
+                t = type(r[0])
+            else:
+                t = type(r[0][0])
+
+        if t == np.float64 or t == float:
```

Comment:
This would be better with `np.issubdtype(t, np.floating)`, and `np.issubdtype(t, np.integer)` below. This works for both np dtypes and Python float and int!